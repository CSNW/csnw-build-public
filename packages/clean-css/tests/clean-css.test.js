const { join } = require('path');
const { Files } = require('@csnw-build/pipeline');

const cleanCss = require('../')({});
const cwd = join(__dirname, 'fixtures');

test('pipeline', async () => {
  const process = await cleanCss.pipeline({
    cwd,
    cache: false,
    sourcemaps: true,
    mode: 'production'
  });
  const files = await Files.from('**/*', { cwd });

  const result = await process(files);
  expect(result).toMatchSnapshot();
});
