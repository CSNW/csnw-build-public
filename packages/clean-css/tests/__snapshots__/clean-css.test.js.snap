// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pipeline 1`] = `
Object {
  "a.css": Object {
    "baseDir": "C:/dev/csnw-build/packages/clean-css/tests/fixtures",
    "contents": ".name{color:red;font-size:1em}",
    "cwd": "C:/dev/csnw-build/packages/clean-css/tests/fixtures",
    "path": "C:/dev/csnw-build/packages/clean-css/tests/fixtures/a.css",
    "sourceMap": Object {
      "file": "a.css",
      "mappings": "AAAA,MACE,MAAO,IACP,UAAW",
      "names": Array [],
      "sources": Array [
        "a.css",
      ],
      "sourcesContent": Array [
        ".name {
  color: red;
  font-size: 1em;
}
",
      ],
      "version": 3,
    },
  },
}
`;
